### PR TITLE
small changes to profiles and additional tests

### PR DESCRIPTION
**Pull request by ahuang314**
1. Revisited some profiles and changed some functions to staticmethods whenever possible so that functions do not have to be recompiled each time a new instance of the class is created. Also restored some missing jit decorators.
2. There was a major bug in the PseudoJaffe profiles when Ra and Rs were swapped. Previously, Ra would be updated first, then Rs updated second, but Rs would not correctly update since it uses the updated Ra in its conditional check. This has been fixed so that they now update at the same time. Note that autodifferentiation works correctly when swapping variables like this.
3. I procrastinated on doing more important things and instead improved the numerical integration method done in the Gaussian profile. We now use Weddle's rule instead of trapezoidal integration for much better accuracy without sacrificing too much performance.
4. I added additional tests in test_likelihood_module that checks whether autodifferentiation works on every currently implemented deflector and source profile.
5. Changed the Shapelets profile to make it compatible with autodifferentiation
6. I added a notebook for comparing the performance between jaxtronomy and lenstronomy for computing ray tracing and surface brightness. Also updated the readme

### CODE REVIEW

File: jaxtronomy/LensModel/Profiles/gaussian.py [hunk 1]
```diff
@@ -53,44 +42,51 @@ def function(self, x, y, amp, sigma, center_x=0, center_y=0):
...
-            x_left = dx * i
-            x_right = dx * (i + 1)
-            y_left = jnp.where(x_left == 0, 0, (1 - jnp.exp(-c * x_left**2)) / x_left)
-            y_right = (1 - jnp.exp(-c * x_right**2)) / x_right
-            sum += (y_left + y_right) / 2 * dx
+        subinterval_widths = r / 100.0
+        coeffs = np.array([1, 5, 1, 6, 1, 5, 1], dtype=float) / 20
+
+        def weddles_rule(i, sum):
+            x = (jnp.ones((7, len(r))) * subinterval_widths).T * (
```
Comment (sibirrer) [inline jaxtronomy/LensModel/Profiles/gaussian.py:76]: can you document what this definition is doing?

File: jaxtronomy/LensModel/Profiles/gaussian.py [hunk 2]
```diff
@@ -154,66 +153,71 @@ def density_2d(self, x, y, amp, sigma, center_x=0, center_y=0):
...
         :param R: projected radius
         :param amp: 3d amplitude of Gaussian
         :param sigma: standard deviation of Gaussian
         """
         sigma_x, sigma_y = sigma, sigma
-        amp2d = amp / (jnp.sqrt(jnp.pi) * jnp.sqrt(sigma_x * sigma_y * 2))
+        amp2d = amp / (np.sqrt(np.pi) * jnp.sqrt(sigma_x * sigma_y * 2))
         c = 1.0 / (2 * sigma_x * sigma_y)
-        return amp2d * 2 * jnp.pi * 1.0 / (2 * c) * (1.0 - jnp.exp(-c * R**2))
+        return amp2d * 2 * np.pi * 1.0 / (2 * c) * (1.0 - jnp.exp(-c * R**2))
```
Comment (sibirrer) [inline jaxtronomy/LensModel/Profiles/gaussian.py:173]: just checking whether this is intended to use np and not jnp

**Review by sibirrer — CHANGES_REQUESTED** [review]
Thank you very much! I have a minor request and a comment just to confirm

File: notebooks/performance_comparison_with_lenstronomy.ipynb [hunk 3]
```diff
@@ -0,0 +1,321 @@
+{
```
Comment (sibirrer) [inline notebooks/performance_comparison_with_lenstronomy.ipynb:1]: <p>do you also compare the "EPL" model? I only see EPL_Q_PHI and EPL_NUMBA</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lenstronomy/JAXtronomy/pull/40/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

File: jaxtronomy/LensModel/Profiles/gaussian.py (hunk 2 above)
Comment (ahuang314) [reply jaxtronomy/LensModel/Profiles/gaussian.py:173]: I changed it to np.pi because numpy functions are executed at compile time (sometimes resulting in speedup if it does not have to be continuously evaluated at runtime), but it turns out jnp.pi and np.pi are both known at compile time anyways, so this makes no noticeable difference

File: notebooks/performance_comparison_with_lenstronomy.ipynb (hunk 3 above)
Comment (ahuang314) [reply notebooks/performance_comparison_with_lenstronomy.ipynb:1]: The comparison is between EPL in jaxtronomy with EPL_NUMBA in lenstronomy. There is not much use in comparing with the regular EPL in lenstronomy I think.

Comment (ahuang314) [reply notebooks/performance_comparison_with_lenstronomy.ipynb:1]: I will update the text to be clearer about what is being compared

**Review by sibirrer — APPROVED** [review]
Looks good
